src/lib.rs

```diff
@@ -1,25 +1,45 @@
-use std::io::BufRead;
+use std::path::Path;
 
+use rayon::iter::ParallelBridge;
+use rayon::prelude::*;
+
+mod chunks;
 mod counter;
 mod key_finder;
 
+pub use chunks::chunks;
 pub use counter::{Counter, KeyCount};
 pub use key_finder::KeyFinder;
 
-pub fn top_few_from_stream<T: BufRead>(
-    reader: T,
+fn sum(mut l: Counter, r: Counter) -> Counter {
+    for (key, count) in r.counts().into_iter() {
+        l.add(&key, count);
+    }
+    l
+}
+
+pub fn top_few_from_stream(
+    path: &Path,
     kf: &KeyFinder,
     num: usize,
-) -> Result<Vec<KeyCount>, anyhow::Error> {
-    let mut counter = Counter::new(num);
-    let mut s = String::new();
-    for ln in reader.lines() {
-        s.clear();
-        let ln = ln?;
-        if let Ok(key) = kf.key(&ln, &mut s) {
-            counter.add(key)
-        }
-    }
+) -> anyhow::Result<Vec<KeyCount>> {
+    let total = chunks(path, 1 << 26)?
```

Comment:
I still don't understand what the `1 << 26` value means, semantically. Please make it a named constant. It's also unclear why you use `1 << 26` rather than `1 << 25` or `1 << 27` (this could be explained in a comment). Wouldn't it make more sense to specify the number of chunks to aim for rather than a specific chunk size, and then use the number of (v)CPU's available by default?

Reply:
With rayon's work stealing implementation you want to have more chunks (10x) than CPUs to make the CPU utilization more even. I'll change the implementation to refect this.